packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1103,6 +1104,46 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
   /// Override in subclasses with children and call the visitor for each child.
   void visitChildren(RenderObjectVisitor visitor) { }
 
+  /// Capture an image of the current state of the render object and its
+  /// children.
+  ///
+  /// The returned [ui.Image] has uncompressed raw bytes in the dimensions of
+  /// the render object, multiplied by the [pixelRatio].
+  ///
+  /// The property [isRepaintBoundary] must be true for [toImage] to work, and
+  /// the render object must not need painting (i.e. [debugNeedsPaint] must be
+  /// false).
+  ///
+  /// The [pixelRatio] describes the scale between the logical pixels and the
+  /// size of the output image. It is independent of the
+  /// [window.devicePixelRatio] for the device, so specifying 1.0 (the default)
+  /// will give you a 1:1 mapping between logical pixels and the output pixels
+  /// in the image.
+  ///
+  /// See also:
+  ///
+  ///  * [ui.Scene.toImage] for more information about the image returned.
+  Future<ui.Image> toImage({double pixelRatio: 1.0}) async {
+    assert(isRepaintBoundary, 'You can only call toImage for RenderObjects that are repaint boundaries.');
+    assert(!_needsPaint);
+    final ui.SceneBuilder builder = new ui.SceneBuilder();
+    Matrix4 transform = new Matrix4.diagonal3Values(pixelRatio, pixelRatio, 1.0);
+    transform = transform.multiplied(
+      new Matrix4.translationValues(-layer.offset.dx, -layer.offset.dy, 0.0),
+    );
+    builder.pushTransform(transform.storage);
+    layer.addToScene(builder, Offset.zero);
+    final ui.Scene scene = builder.build();
+    try {
+      return await scene.toImage(
+        (pixelRatio * paintBounds.width).round(),
```

Comment:
`paintBounds` is a bit of a dubious choice because it's just advisory really.
If you move this to RenderRepaintBoundary, you can just use `size`, which is much more authoritative.

Reply:
Done.